Tag creation via manual workflow failed

Run git config --global user.name "brainupgrade-in"
remote: Permission to brainupgrade-in/copilot-weather.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/brainupgrade-in/copilot-weather/': The requested URL returned error: 403
Error: Process completed with exit code 12[8](https://github.com/brainupgrade-in/copilot-weather/actions/runs/10304204097/job/28522061368#step:3:9).

--- a/.github/workflows/create-tag.yaml
+++ b/.github/workflows/create-tag.yaml
@@ -14,6 +14,9 @@ jobs:
   create_tag:
     runs-on: ubuntu-latest
 
+    permissions:
+      contents: write
+
     steps:
     - name: Check out repository
       uses: actions/checkout@v2
@@ -24,3 +27,5 @@ jobs:
         git config --global user.email "${{ github.actor }}@users.noreply.github.com"
         git tag -a ${{ github.event.inputs.tag_name }} -m "${{ github.event.inputs.tag_message }}"
         git push origin ${{ github.event.inputs.tag_name }}
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

